src/plugin.ts

```diff
@@ -818,4 +825,63 @@ export class MongoDBOIDCPluginImpl implements MongoDBOIDCPlugin {
     }
     this.logger.emit('mongodb-oidc-plugin:destroyed');
   }
+
+  public async userInfo<
+    UserInfo extends Record<string, unknown> = Record<string, never>,
+    Address extends Record<string, unknown> = Record<string, never>
+  >(
+    serverMetadata: IdPServerInfo,
+    options: { signal?: OIDCAbortSignal } = {}
+  ): Promise<UserinfoResponse<UserInfo, Address>> {
+    return await withAbortCheck(options.signal, async ({ signalPromise }) => {
+      return await Promise.race([
+        (async () => {
+          const authState = this.getAuthState(serverMetadata);
+          const { client } = await this.getOIDCClient(authState);
+          return await client.userinfo<UserInfo, Address>(
+            authState.currentTokenSet?.set ?? ({} as TokenSet)
```

Comment:
Should this throw if there is no current token set?

Reply:
I don't think so. I am not sure why oidc-plugin should do this validation and not backend that this is communicating with